File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s){
        StringBuilder sb = new StringBuilder();
        String s1,s2;
        s=s.toLowerCase();
        for(int i=0;i<s.length();i++)
        {
            char c = s.charAt(i);
            if(Character.isLetterOrDigit(c))
            {
                sb.append(c);
            }
        }
        s1=sb.toString();
        s2 = sb.reverse().toString();
       return s1.equals(s2);
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s=s.lower()
        i=0
        j=len(s)-1
        while(i<j):
            if not s[i].isalnum():
                i=i+1
            elif not s[j].isalnum():
                j=j-1
            elif s[i]!= s[j]:
                return False
            else:
                i=i+1
                j=j-1        

        return True        